### PR TITLE
22 Migrate Color Wheel Activity #22

### LINKED ISSUE #22: [Activity] Migrate Color Wheel Activity to Wrapper
**Is your feature request related to a problem? Please describe.**
Migrate https://github.com/codewithher/CC-Library-Lantern/pull/15 to [CC-Lantern](https://github.com/codewithher/CC-Lantern)

**Describe the solution you'd like**
Use the provided wrapper library to further simplify the activity.

**Describe alternatives you've considered**
- N/A

**Additional context**

### PR DESCRIPTION
**Pull request by innaamogolonova**
**Why:**

Closes the following issues (all PRs must close an issue):

- closes #22 

**What's being changed**

Simplifies the rainbow() method in act3 examples to implement Lantern.h library, making the function easier to read. 

**Testing instructions**

The easiest way to test is to run the rainbow method in activity 3 which double as testing files. 

You can verify that the code works appropriately by visually inspecting the colors that result from the activity function calls.

- [x] The LED goes through the complete color spectrum (red to green to blue back to red)
- [x] The color change is gradual and smooth 
- [x] The rainbow lasts the specified "seconds" amount of time
- [x] Change the number of seconds and make sure the rainbow lasts appropriate amount of time 

**Check off the following**

- [x] I have assigned @totally-not-frito-lays as a reviewer

### CODE REVIEW

**Review by totally-not-frito-lays — CHANGES_REQUESTED** [review]
Just asking for a bit more clarity. The code works well as is really enjoyable to watch.

Something interesting I noted is that when giving invalid input (ie. negative numbers) such as `rainbow(-5)` the behavior here is different than from `CC-Library-Lantern`. In `CC-Library-Lantern` the behavior was rapidly changing colors. In `CC-Lantern` (this repo), the behavior runs slowly, starting at orange.

This may be a quirk of our implementation of the `wait()` wrapper function. I'll open up a ticket for this.

**Comment by totally-not-frito-lays** [comment]
> Just asking for a bit more clarity. The code works well as is really enjoyable to watch.
> 
> Something interesting I noted is that when giving invalid input (ie. negative numbers) such as `rainbow(-5)` the behavior here is different than from `CC-Library-Lantern`. In `CC-Library-Lantern` the behavior was rapidly changing colors. In `CC-Lantern` (this repo), the behavior runs slowly, starting at orange.
> 
> This may be a quirk of our implementation of the `wait()` wrapper function. I'll open up a ticket for this.

Opened ticket #25 for this. Bounds handling inside this function is the easiest way to resolve this for now but it may be worth some exploration to see if this occurs for other functions.

Just to clarify, this PR does not need to resolve #25.

**Comment by innaamogolonova** [comment]
Addressed the changes requested by: 
- commenting out other functions
- adding a bounds check
- adding an explanation for 765 number (both in code and on notion)

File: examples/act3-functions/act3-functions.ino [hunk 1]
```diff
@@ -12,71 +12,38 @@ void setup() {
...
+  int Red = 0; 
+  int Green = 0; 
+  int Blue = 0;
 
-  for (int step = 0; step < numSteps; step++) {
-    Red -= 50;
-    lantern.setColor(seconds, lantern.color(Red, Green, Blue));
+  for (Red = 255; Red > 0; Red --) {
+    Green += 1; 
+    lantern.setColor(seconds / 765, lantern.color(Red, Green, Blue));
```
[the remark this answers is not in the document]
Comment (totally-not-frito-lays) [reply examples/act3-functions/act3-functions.ino:40]: @innaamogolonova I think we can communicate some of what the comment is saying into code.

ie.
```ino
const int steps = 255;  // Number of steps for the gradient
const int num_led = 3;  // Number of colors (RGB)
const int total_steps = steps * num_led;  // Total steps in the gradient
const float time_per_color = seconds / total_steps;  // Time for each color step

for (int Red = 255; Red > 0; Red--) {
    Green += 1;
    lantern.setColor(time_per_color, lantern.color(Red, Green, Blue));
}

**Review by totally-not-frito-lays — APPROVED** [review]
Approving this PR! I just made the code a little more readable in the example :)

**Pull request merged**